packages/create-cloudflare/src/common.ts

```diff
@@ -36,49 +36,44 @@ export const validateProjectDirectory = (
 	relativePath: string,
 	args: Partial<C3Args>
 ) => {
-	// Validate that the directory is non-existent or empty
-	const path = resolve(relativePath);
-	const existsAlready = existsSync(path);
-	const isEmpty = existsAlready && readdirSync(path).length === 0; // allow existing dirs _if empty_ to ensure c3 is non-destructive
-
-	if (existsAlready && !isEmpty) {
-		return `Directory \`${relativePath}\` already exists and is not empty. Please choose a new name.`;
-	}
-
-	// Ensure the name is valid per the pages schema
 	// Skip this if we're initializing from an existing workers script, since some
 	// previously created workers may have names containing capital letters
-	if (!args.existingScript) {
-		const projectName = basename(path);
-		const invalidChars = /[^a-z0-9-]/;
-		const invalidStartEnd = /^-|-$/;
+	if (args.existingScript) return;
 
-		if (projectName.match(invalidStartEnd)) {
-			return `Project names cannot start or end with a dash.`;
-		}
+	// Ensure the name is valid per the pages schema
+	const path = resolve(relativePath);
+	const projectName = basename(path);
+	const invalidChars = /[^a-z0-9-]/;
+	const invalidStartEnd = /^-|-$/;
 
-		if (projectName.match(invalidChars)) {
-			return `Project names must only contain lowercase characters, numbers, and dashes.`;
-		}
+	if (projectName.match(invalidStartEnd)) {
+		return `Project names cannot start or end with a dash.`;
+	}
 
-		if (projectName.length > 58) {
-			return `Project names must be less than 58 characters.`;
-		}
+	if (projectName.match(invalidChars)) {
+		return `Project names must only contain lowercase characters, numbers, and dashes.`;
+	}
+
+	if (projectName.length > 58) {
+		return `Project names must be less than 58 characters.`;
 	}
 };
 
 export const setupProjectDirectory = (args: C3Args) => {
-	// Crash if the directory already exists
 	const path = resolve(args.projectName);
 	const err = validateProjectDirectory(path, args);
 	if (err) {
 		crash(err);
 	}
 
+	// Delete the target directory path. At this point, the
+	// user has already confirmed the overwrite, if it exists
+	rmSync(path, { recursive: true, force: true });
```

Comment:
I'm not super familiar with this code, so this is a drive-by review, but this feels very dangerous to me. Someone could run this in `/`, or `C:/` with `--overwrite` and then end up with a bunch of their filesystem being wiped.

Reply:
I'm going to change it back to the previous strategy of failing on non-empty directories with the exception of an allow list, but for posterity: 

> Someone could run this in /, or C:/ with --overwrite and then end up with a bunch of their filesystem being wiped if I'm understanding correctly?

Correct, if you ran `npm create cloudflare / -- --overwrite, it would wipe the filesystem. Just as it would if you ran `rm -rf /`. Does this seem like a common, realistic use case that we should account for? 

> Even with a confirmation, that feels very heavy-handed and not something I've personally experienced with other tooling.

Here's a couple of examples of similar tools that do the same thing:
- [create-solid](https://github.com/solidjs/solid-start/blob/main/packages/create-solid/cli/index.js#L300)
- [create-vue](https://github.com/vuejs/create-vue/blob/main/index.ts#L316 )

Reply:
Thanks for the insight @jculvey. I think the previous strategy makes the most sense.

As for those other packages, if I'm reading them correctly from a very brief investigation:

- `create-solid` seems to only remove the `tempTemplate` dir inside your current working directory, which they define as `path.join(target, ".solid-start")`.
- and `create-vue` seems to explicitly check if they enter a `.`, requires a `--force`, and even re-prompts them that they're about to remove everything in that directory